packages/core/src/browser/preferences/preference-proxy.ts

```diff
@@ -21,69 +22,76 @@ export interface PreferenceChangeEvent<T> {
-                if (validatePreference(schema, {
-                    [e.preferenceName]: e.newValue
-                })) {
+                if (validate(e.preferenceName, e.newValue)) {
                     onPreferenceChangedEmitter.fire(e);
                 } else {
-                    // Fire the default preference
                     onPreferenceChangedEmitter.fire({
                         preferenceName: e.preferenceName,
                         newValue: configuration[e.preferenceName]
                     });
                 }
             } else {
-                // TODO If it's deleted, fire the default preference
-                onPreferenceChangedEmitter.fire(e);
+                onPreferenceChangedEmitter.fire({
+                    preferenceName: e.preferenceName,
+                    newValue: configuration[e.preferenceName],
+                    oldValue: e.oldValue
+                });
             }
         }
     }));
-    // tslint:disable-next-line:no-any
+    const unsupportedOperation = (_: any, property: string) => {
+        throw new Error('Unsupported operation');
+    };
     return new Proxy(configuration as any, {
-        get: (_, p: string) => {
-            if (p in schema.properties) {
-                if (p in configuration) {
-                    const preference = preferences.get(p, configuration[p]);
-                    if (validatePreference(schema, {
-                        [p]: preference
-                    })) {
-                        return preference;
-                    } else {
-                        return configuration[p];
-                    }
+        get: (_, property: string) => {
+            if (property in configuration) {
+                const preference = preferences.get(property, configuration[property]);
+                if (validate(property, preference)) {
+                    return preference;
                 } else {
-                    return schema.properties[p].default || undefined;
+                    return configuration[property];
                 }
             }
-            if (p === 'onPreferenceChanged') {
+            if (property === 'onPreferenceChanged') {
                 return onPreferenceChangedEmitter.event;
             }
-            if (p === 'dispose') {
+            if (property === 'dispose') {
                 return () => toDispose.dispose();
             }
-            if (p === 'ready') {
+            if (property === 'ready') {
                 return () => preferences.ready;
             }
-            throw new Error('unexpected property: ' + p);
-        }
+            throw new Error('unexpected property: ' + property);
+        },
+        set: unsupportedOperation,
+        deleteProperty: unsupportedOperation,
+        defineProperty: unsupportedOperation
     });
 }
 
+function createConfiguration<T extends Configuration>(schema: PreferenceSchema): T {
```

Comment:
That's the idea I had in mind at first :)

Reply:
that's also that I tried to explain, sorry for the misunderstanding 😁